SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp

```diff
@@ -197,6 +198,43 @@ TraceKey USpatialLatencyTracer::ReadTraceFromSchemaObject(Schema_Object* Obj, co
 	return InvalidTraceKey;
 }
 
+TraceKey USpatialLatencyTracer::ReadTraceFromSpatialPayload(const FSpatialLatencyPayload& Payload)
+{
+	FScopeLock Lock(&Mutex);
+
+	if (Payload.TraceId.Num() != sizeof(improbable::trace::TraceId))
+	{
+		UE_LOG(LogSpatialLatencyTracing, Warning, TEXT("Payload TraceId does not contain the correct number of trace bytes. %d found"), Payload.TraceId.Num());
+		return InvalidTraceKey;
+	}
+
+	if (Payload.SpanId.Num() != sizeof(improbable::trace::SpanId))
+	{
+		UE_LOG(LogSpatialLatencyTracing, Warning, TEXT("Payload TraceId does not contain the correct number of span bytes. %d found"), Payload.SpanId.Num());
+		return InvalidTraceKey;
+	}
+
+	improbable::trace::TraceId _TraceId;
+	memcpy(&_TraceId[0], Payload.TraceId.GetData(), sizeof(improbable::trace::TraceId));
```

Comment:
I wonder if it might be advantageous to first create the improbable::trace::SpanContext and then memcpy into its properties.

improbable::trace::SpanContext DestContext();
memcpy(&DestContext._TraceId[0], Payload.TraceId.GetData(), sizeof(improbable::trace::TraceId));
memcpy(&DestContext._SpanId[0], Payload.SpanId.GetData(), sizeof(improbable::trace::SpanId));

Reply:
If we prefer this method and it is possible we might want to also modify USpatialLatencyTracer::ReadTraceFromSchemaObject as it has similar logic.

Reply:
Unfortunately those members are private in SpanContext (and owned by workersdk) - i'll wrap it up though as we do it in multiple places

Reply:
"Unfortunately those members are private" feelsbadman...